lib/coverband/collectors/coverage.rb

```diff
@@ -74,14 +71,12 @@ def ready_to_report?
       end
 
       def get_new_coverage_results
-        coverage_results = nil
-        @semaphore.synchronize { coverage_results = new_coverage(::Coverage.peek_result.dup) }
-        coverage_results
+        @semaphore.synchronize { new_coverage(::Coverage.peek_result.dup) }
```

Comment:
Minor note but when we are using background thread, we should be able to avoid the `@semaphore` since there should always be 1 thread.

Reply:
yep thought about it but since I want to keep middleware option around for now we need to keep it